mage_ai/data_preparation/models/pipeline.py

```diff
@@ -1013,6 +1013,14 @@ async def update(self, data, update_content=False):
 
             cache = await TagCache.initialize_cache()
 
+            # Update tag cache if tag was removed from pipeline
+            tags_cached = cache.get_tags_for_specific_pipeline(self)
+            tags_removed = set(tags_cached) - set(self.tags)
+            if len(tags_removed) > 0:
+                for tag_uuid in tags_removed:
+                    if old_uuid is None:
+                        cache.remove_pipeline(tag_uuid, self.uuid)
```

Comment:
I noticed fetching pipelines with tag filters in the pipelines list wasn't getting updated immediately when a tag was removed from a pipeline, so this fixes that. Calling the `remove_pipeline` method also clears the `pipeline_uuid_to_tags_mapping` cache, so that an outdated cache would not be used for the Pipeline Runs table.

For example in the screenshot, the filter for tags with the `integration` tag was applied, but a pipeline (`cloud_postgres_to_mysql`) that no longer has that tag was still included. You can see that it doesn't have the `integration` tag in the "Tags" column.
![image](https://github.com/mage-ai/mage-ai/assets/78053898/4366d47b-a762-4612-beed-be3e664bfc1e)